app/models/form526_submission.rb

```diff
@@ -68,6 +60,32 @@ def start
     start_evss_submission_job
   end
 
+  def start_rrd_job(submission_id, rrd_processor_class, backup_processor_class = nil)
+    workflow_batch = Sidekiq::Batch.new
+    workflow_batch.on(
+      :success,
+      'Form526Submission#rrd_complete_handler',
+      'submission_id' => submission_id
+    )
+    if backup_processor_class
+      workflow_batch.on(
+        :death,
+        'Form526Submission#rrd_processor_failed_handler',
+        'submission_id' => submission_id
+      )
+    end
```

Comment:
If a `backup_processor_class` is provided and the job for the `rrd_processor_class` fails, then call `rrd_processor_failed_handler`, which will restart the RRD job except using the `backup_processor_class`.